django/contrib/auth/base_user.py

```diff
@@ -131,14 +131,15 @@ def has_usable_password(self):
         """
         return is_password_usable(self.password)
 
-    def get_session_auth_hash(self):
+    def get_session_auth_hash(self, secret=None):
```

Comment:
`get_session_auth_hash()` is a [documented](https://docs.djangoproject.com/en/4.1/topics/auth/customizing/#django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash) API so I'm afraid we cannot freely change it's signature :disappointed: What about adding :thinking::
```python
def get_session_auth_fallback_hash(self):
    key_salt = "django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash"
    for fallback_secret in settings.SECRET_KEY_FALLBACKS:
        yield return salted_hmac(
            key_salt, self.password, secret=fallback_secret, algorithm="sha256"
        ).hexdigest()
```
or too be more DRY:
```python
def _get_session_auth_hash(self, secret=None):
    key_salt = "django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash"
    return salted_hmac(
        key_salt,
        self.password,
        secret=secret,
        algorithm="sha256",
    ).hexdigest()

def get_session_auth_hash(self):
    """
    Return an HMAC of the password field.
    """
    self._get_session_auth_hash()    

def get_session_auth_fallback_hash(self):
    for fallback_secret in settings.SECRET_KEY_FALLBACKS:
        yield self._get_session_auth_hash(secret=fallback_secret)
```
New method should be [documented](https://docs.djangoproject.com/en/4.1/topics/auth/customizing/#django.contrib.auth.models.AbstractBaseUser) with a `versionadded:: 4.1.8` annotation.

Reply:
Sounds like a plan! Thanks a lot for the suggestion 👍